scipy/sparse/_base.py

```diff
@@ -112,11 +112,15 @@ def _lil_container(self):
         from ._lil import lil_array
         return lil_array
 
-    def __init__(self, maxprint=MAXPRINT):
+    def __init__(self, arg1, maxprint=MAXPRINT):
         self._shape = None
         if self.__class__.__name__ == '_spbase':
             raise ValueError("This class is not intended"
                              " to be instantiated directly.")
+        if isinstance(self, sparray) and np.isscalar(arg1):
```

Comment:
Implementing the suggestion from our call means that `_spbase` now needs to check the first argument which resulted in this change.  Not sure if the lack of arguments in `_spbase` was a choice or not, but maybe there's a way around this?

Reply:
Hmmm... Good point about lack of arguments being a choice or not.
It looks like setting `maxprint` actually doesn't work for all the subclasses that override `__init__`. So currently we can't choose a non-default maxprint value for most of the formats.

That could be fixed by adding the maxprint parameter to `__init__` methods throughout.  Or by adding `**kwargs` parameters that get passed through to the superclass `__init__`.  The second option is harder to read for someone trying to figure out what all the options are.

And this could be fixed in a separate PR. Let me know if you prefer that and I'll open a separate issue.

Reply:
Yeah, the `maxprint` functionality has been inaccessible for a while, unfortunately. We should clean that up in a later PR.